### PR TITLE
Fix to wire of wire

### PR DESCRIPTION
**Pull request by imeckler**
This fixes the wire and graphql representations of parties transactions and the conversions between them

### CODE REVIEW

File: src/lib/mina_base/party.ml [hunk 1]
```diff
@@ -508,6 +515,17 @@ module Account_precondition = struct
     end
   end]
 
+  let gen : t Quickcheck.Generator.t =
+    Quickcheck.Generator.variant3 Zkapp_precondition.Account.gen
+      Account.Nonce.gen Unit.quickcheck_generator
+    |> Quickcheck.Generator.map ~f:(function
+         | `A x ->
```
Comment (psteckler) [inline src/lib/mina_base/party.ml:522]: better names than `x`?

File: src/lib/mina_base/party.ml [hunk 2]
```diff
@@ -1058,6 +1353,20 @@ module Fee_payer = struct
...
 
+  let gen : t Quickcheck.Generator.t =
+    let open Quickcheck.Let_syntax in
+    let%map body = Body.Fee_payer.gen in
+    let authorization = Signature.dummy in
+    { body; authorization }
+
+  let quickcheck_generator : t Quickcheck.Generator.t = gen
+
+  let quickcheck_obserber : t Quickcheck.Observer.t =
```
Comment (psteckler) [inline src/lib/mina_base/party.ml:1364]: misspelling

File: src/lib/mina_base/parties.ml [hunk 3]
```diff
@@ -921,15 +1150,18 @@ include Codable.Make_base58_check (Stable.Latest)
...
-    Call_forest.to_list ps
+  and to_parties_with_depth (ps : other_parties) : Party.Graphql_repr.t list =
+    ps
+    |> Call_forest.to_parties_list_map ~f:(fun ~depth p ->
+           Party.to_graphql_repr ~call_depth:depth p )
   in
   let open Fields_derivers_zkapps.Derivers in
-  let inner = (list @@ Party.deriver @@ o ()) @@ o () in
+  let inner = (list @@ Party.Graphql_repr.deriver @@ o ()) @@ o () in
   iso ~map:of_parties_with_depth ~contramap:to_parties_with_depth inner obj
```
Comment (psteckler) [inline src/lib/mina_base/parties.ml:1165]: can there be a test that demonstrates the isomorphism?

**Review by psteckler — APPROVED** [review]
couple of nits, otherwise good

**Comment by mitschabaude** [comment]
Heads up: I fixed the JS side of things here https://github.com/MinaProtocol/mina/pull/11057/commits/499e97a88aa1834dc63c7b4564ea39ba2b681e51

I also think that, now that `Party.Graphql_repr.t` / `Party.Body.Graphql_repr.t` are the proper graphql / JS representations of this type, we should remove the original derivers on `Party`, `Party.Body`, to avoid confusion and ambiguity about how to serialize those types. I went ahead and made this change here: https://github.com/MinaProtocol/mina/pull/11057/commits/668499f4cbaa6b2f91fada9c1b52fc50c22ae95d -- feel free to revert that last commit if you didn't want this

**Review by mitschabaude — APPROVED** [review]

**Review by mrmr1993 — APPROVED** [review]

**Comment by psteckler** [comment]
I fixed the replayer build here. There's a remaining issue with the linters because of the use of `deriving quickcheck`. When #11068 is merged, and `develop` merged into this branch, that should fix the issue.

**Pull request merged**